okteto/okteto
# Feature Flag to Opt-In to Static Tokens/Existing Flow

# Proposed changes

Fixes https://github.com/okteto/app/issues/6886

This branch requires the following PRs to be merged first:
- https://github.com/okteto/okteto/pull/3891
- https://github.com/okteto/okteto/pull/3887

## Description

Adding feature flag via env var `OKTETO_USE_STATIC_KUBETOKEN`. This allows opt-in for using the static tokens.

## How to test

- Using https://github.com/okteto/movies
- Set `export OKTETO_USE_STATIC_KUBETOKEN=true`
- Run `okteto deploy`
- Notice the warning
- Set `export OKTETO_USE_STATIC_KUBETOKEN=false`
- Run `okteto deploy`
- Check the warning is not displayed anymore

Then, to verify the `okteto kubeconfig` change:

- Back up your `~/.kube/config` (just in case)
- Set `export OKTETO_USE_STATIC_KUBETOKEN=true`
- Run `okteto kubeconfig`
- Check your `~/.kube/config` and notice how you are using a token
- Set `export OKTETO_USE_STATIC_KUBETOKEN=false`
- Run `okteto kubeconfig`
- Check your `~/.kube/config` and notice how you are using the `exec` and the token has been removed
